collectd/collectd.go

```diff
@@ -0,0 +1,64 @@
+package collectd
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"math"
+	"time"
+
+	"github.com/kimor79/gollectd"
+)
+
+var ErrTimeOutOfBounds = errors.New("The time is to large to store")
+
+//TODO corylanou: This needs to be made a public `main.Point` so we can share this across packages.
+type Metrics []Metric
+type Metric struct {
+	Name      string
+	Tags      map[string]string
+	Value     interface{}
+	Timestamp time.Time
+}
+
+func Unmarshal(data *gollectd.Packet) (Metrics, error) {
```

Comment:
in the words of @extemporalgenome

the struct contains a map, which can grow during generation, making extra GC objects to check, and the empty interface they're using (which due to being boxed in a deeper function and returned up the stack, will force the static analyzer to heapify it)